core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -57,8 +57,6 @@ public class TableMetadata implements Serializable {
   static final int INITIAL_SORT_ORDER_ID = 1;
   static final int INITIAL_SCHEMA_ID = 0;
   static final int INITIAL_ROW_ID = 0;
-  static final boolean DEFAULT_ROW_LINEAGE = false;
-  static final int MIN_FORMAT_VERSION_ROW_LINEAGE = 3;
```

Comment:
I still kind of like having this as a CONSTANT at the top of the file so we can be clear when we are gating something it's because of row_lineage and not generic v3 things. But either way is fine with me